src/components/AddTask/AddTask.tsx

```diff
@@ -0,0 +1,82 @@
+import "./AddTask.css";
+import addButton from "../../assets/add-button.svg";
+import TaskListContainer from "../Task/TaskList/TaskListContainer";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+interface Task {
+  id: number;
+  name: string;
+  completed: boolean;
+}
+
+interface AddTaskProps {
+  tasks: Task[];
+  addTask: (task: Task) => void;
+}
+
+const schema = yup
```

Comment:
validation should be in seprate file/folder and use Schema Builder while using hook form